models/weight_recorder.cpp

```diff
@@ -59,81 +52,42 @@ nest::weight_recorder::weight_recorder( const weight_recorder& n )
 {
 }
 
+// We must initialize senders and targets here with empty NCs because
+// they will be returned by get_status()
 nest::weight_recorder::Parameters_::Parameters_()
-  : senders_()
-  , targets_()
+  : senders_( new NodeCollectionPrimitive() )
+  , targets_( new NodeCollectionPrimitive() )
 {
 }
 
 void
-nest::weight_recorder::Parameters_::get( DictionaryDatum& d ) const
+nest::weight_recorder::Parameters_::get( Dictionary& d ) const
 {
-  if ( senders_.get() )
-  {
-    ( *d )[ names::senders ] = senders_;
-  }
-  else
-  {
-    ArrayDatum ad;
-    ( *d )[ names::senders ] = ad;
-  }
-  if ( targets_.get() )
-  {
-    ( *d )[ names::targets ] = targets_;
-  }
-  else
-  {
-    ArrayDatum ad;
-    ( *d )[ names::targets ] = ad;
-  }
+  d[ names::senders ] = senders_;
+  d[ names::targets ] = targets_;
 }
 
 void
-nest::weight_recorder::Parameters_::set( const DictionaryDatum& d )
+nest::weight_recorder::Parameters_::set( const Dictionary& d )
 {
-  if ( d->known( names::senders ) )
+  auto get_or_create_nc = [ &d ]( NodeCollectionPTR& nc, const std::string& key )
```

Comment:
Shouldn't this be called just `get_nc` instead of `get_or_create_nc`, as an empty NC sis already created in the constructor?